### PR TITLE
New blog post: A simple TDD environment in Haskell

### PR DESCRIPTION
**Pull request by Gryff**
Feedback is much appreciated.

### CODE REVIEW

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+date: 2016-05-25 00:01:00 +00:00
+author: Liam Griffin
+image:
+    src: /assets/img/custom/blog/2016-05-25-haskell.svg
+tags:
+- tdd
+- haskell
+---
+
+I recently tried the bowling kata in Haskell ([finished implementation here](https://github.com/Gryff/bowling-kata)), which caused me to find out how set up my environment to comfortably do TDD. Hopefully others might find this helpful to begin their journey with the language. Here’s what I used:
```
Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: `I recently implemented the [bowling kata in Haskell](https://github.com/Gryff/bowling-kata)). In the process, I found out how to set up my environment to comfortably do Test Driven Development. Hopefully, others might find this post helpful to begin their journey with this language. I used the following components:`

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+- Haskell installation: [Haskell Platform](https://www.haskell.org/platform/). This also gives you GHCi which you can use as a REPL and type inspector.
```
Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: explain what REPL is

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- IDE: Any editor would suffice, but I used [Visual Studio Code](https://code.visualstudio.com/) as they have an extension for Haskell that gave me some basic intellisense features.
```
Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: `IntelliSense` is a brand - Link to JetBrains website

Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: `that provides an extension for Haskell with some basic IntelliSense features.`

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+- Test libraries: [Hspec](http://hspec.github.io/) (can be installed from the command line with ```cabal install hspec```).
```
Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: explain cabal

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- Helper libraries: Printf for colourful command line output.
+
+Using an example from some [Codewars](http://www.codewars.com/)’ Haskell tests, I began with this structure for my code:
```
Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: maybe replace with Hspec test if the example was in Hspec

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  it (printf “should return the score for game : %s → %d \n” game score) $
+    scoreGame game `shouldBe` score
+
+main = hspec $ do
+  describe "scoreGame" $ do 
+    testScoreGame "--------------------" 0
+```
+
+
+So to test a function you add a function in your test file, usually the same name with a ‘test’ prefix, which takes as parameters the inputs to your function under test and the expected output. Then using Hspec, which is based on Rspec, you describe what you are testing. As you can see the ‘it’ part is written in the test function. You can of course omit this helper function and write all your tests under ```main = hspec $ do```, which may be nicer if you want to describe in more detail what each individual test is testing.
```
Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: So to test a function, you add a function in your test file, usually the same name with a ‘test’ prefix. This function takes  the inputs to your function under test and the expected output as parameters.

Comment (jukecraft) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: link to Rspec - maybe put this information above when introducing Hspec

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md (hunk 2 above)
Comment (Gryff) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: Maybe just saying 'interactive environment' will be shorter.

File: _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md (hunk 3 above)
Comment (Gryff) [inline _posts/2016-05-25-a-simple-TDD-environment-in-haskell.md]: There's not really an IntelliSense webpage for Visual Studio Code, so I could link to the main VS [IntelliSense page](https://msdn.microsoft.com/en-us/library/hcw1s69b.aspx)

**Comment by Gryff** [comment]
@franziskas Thanks for the feedback, I have made the suggested changes.

**Pull request merged**